src/components/separator/separator.tsx

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+interface SeparatorProps {
+  vertical?: boolean;
+  horizontal?: boolean;
+  size: number;
+}
+
+export const Separator = ({ vertical = false, horizontal = false, size }: SeparatorProps) => {
+  const separatorStyle = {
+    height: vertical ? `${size}px` : undefined,
```

Comment:
Aqui ao invés de atribuir `undefined` caso a prop vertical ou horizontal seja falsa vc poderia atribuir o valor apenas se for verdadeira, usando:
```suggestion
    height: vertical && `${size}px` ,
```

Reply:
Mas, nesse sentido, se a propriedade vertical for false, a expressão também será e causará um erro na propriedade style..

Reply:
Agora que estamos usando o styled components você pode criar um arquivo de estilo para o seu separator ao invés de fazer inline, e com o styled components vc pode adaptar melhor essa lógica acima.
https://styled-components.com/docs/basics#adapting-based-on-props << Esse link pode dar uma clareada melhor.
mas por exemplo no seu arquivo styled:

```suggestion
    export const SeparatorStyle = styled.div<props>`
      ${props => props.vertical && height: vertical };
    `
    
```